apps/wing/src/commands/test/test.ts

```diff
@@ -134,7 +140,20 @@ export async function test(entrypoints: string[], options: TestOptions): Promise
       });
     }
   };
-  await Promise.all(selectedEntrypoints.map(testFile));
+
+  // split the entrypoints to smaller batches according to the command options-
+  // if none- it will be preformed in one batch
+  const batchSize = Number(options.parallel) || selectedEntrypoints.length;
+  const totalBatches = Math.ceil(selectedEntrypoints.length / batchSize);
+  await PromisePool.withConcurrency(batchSize)
+    .for(selectedEntrypoints)
+    .onTaskFinished((_, pool) => {
+      const processed = pool.processedCount();
+      if (processed % batchSize == 0 || processed === selectedEntrypoints.length) {
+        log(`Done processing batch ${Math.ceil(processed / batchSize)}/${totalBatches}`);
+      }
+    })
+    .process(testFile);
```

Comment:
Now that we're not doing batches, I think we can simplify this. I don't think it's important for the user to get feedback about the parallelism since they hopefully won't notice it.

```suggestion
  await PromisePool
    .withConcurrency(options.parallel ?? selectedEntrypoints.length)
    .for(selectedEntrypoints)
    .process(testFile);
```

Also should be consider a sane default max parallelism? Especially for cloud testing currently it's not a good idea to run too many tests at once.

Reply:
ok :) 
I was doing it for being able to see/test if it's working at all - we won't be able to track it otherwise (and also won't be able to write a test, unless spying promisePool)

what would be a good default limit? 10? 20?

Reply:
It should be possible to test by having the tests access something shared as a way to check a lock, like a file. Then setting the parallelism to 1.

Not sure what a good default is, We could try 10.